File: src-server/src/domain/Result.ts
export class Result<T> {

    private readonly code: number;
    private readonly message: string;
    private readonly data?: T;

    constructor(code: number, message: string, data?: T) {
        this.code = code;
        this.message = message;
        this.data = data;
    }

    static success<T>(data?: T): Result<T> {
        return new Result(200, "success", data);
    }

    static fail<T>(message: string): Result<T> {
        return new Result(500, message);
    }

    static noAuth<T>(): Result<T> {
        return new Result(403, "no auth");
    }

}

File: src-java/src/main/java/xyz/esion/knowledgebase/entity/Result.java
package xyz.esion.knowledgebase.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.RequiredArgsConstructor;

import java.io.Serializable;

/**
 * @author Esion
 * @since 2023/10/29
 */
@Data
@RequiredArgsConstructor
@AllArgsConstructor
public class Result<T> implements Serializable {

    private static final long serialVersionUID = 1L;

    private final Integer code;

    private final String message;

    private T data;

    public static <T> Result<T> success(T data) {
        return new Result<>(200, "success", data);
    }

    public static <T> Result<T> success() {
        return new Result<>(200, "success");
    }

    public static Result<Void> fail(String message) {
        return new Result<>(500, message);
    }

}
